### PR TITLE
MaxEnd: find the interval values that contain the max end of the tree

### PR DESCRIPTION
**Pull request by thorfour**
In an attempt to address #7  where I need to find the values that contain the max ending intervals in the tree.

This performs a search on the tree based on the `maxEnd` of the leaves of each sub-tree.

### CODE REVIEW

File: interval/search.go [hunk 1]
```diff
@@ -426,3 +426,37 @@ func (st *MultiValueSearchTree[V, T]) Select(k int) ([]V, bool) {
...
+		return vals, false
+	}
+
+	maxEnd(st.root, st.root.maxEnd, st.cmp, func(n *node[V, T]) {
+		vals = append(vals, n.interval.vals...)
+	})
+	return vals, true
+}
+
+func maxEnd[V, T any](n *node[V, T], searchEnd T, cmp CmpFunc[T], visit func(n *node[V, T])) {
```
Comment (rdleal) [inline interval/search.go:446]: On the definition of the `visit` parameter, the `*node[V, T]` seems pretty descriptive. What do you think of removing the `n` from that definition?
```suggestion
func maxEnd[V, T any](n *node[V, T], searchEnd T, cmp CmpFunc[T], visit func(*node[V, T])) {
```

**Review by rdleal — COMMENTED** [review]
Left a few minor comments related to the documentation of the added code.

**Comment by thorfour** [comment]
Commited your suggestions

**Review by rdleal — APPROVED** [review]
LGTM! Thanks for contributing to the package @thorfour .

**Pull request merged**